flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -267,6 +269,22 @@ public Builder uidPrefix(String newPrefix) {
       return this;
     }
 
+    public Builder customSnapshotMetadata(Map<String, String> metadata) {
+      if (this.customSnapshotMetadata == null) {
+        this.customSnapshotMetadata = Maps.newHashMap();
```

Comment:
For a builder, I'd just initialize the metadata map to this rather than having this `if`.

Also, accessing a field doesn't require `this.`. We only use the prefix when setting a variable, so that it is clear that it is a field being set.